src/imitation_loss/imitation_loss.jl

```diff
@@ -0,0 +1,52 @@
+"""
+    ImitationLoss{L,R,P}
+
+Generic imitation loss: max_y ℓ(y, y̅) + θᵀ(y - y̅) - (Ω(y) - Ω(y̅)).
```

Comment:
The base loss theoretically takes `y` and a target `t_true`, which might also include `theta_true` in addition to `y_true`

Reply:
I pushed a first version of it. Now the base loss can take `t_true` as input, with `t_true` being a names tuple that must contain a field `y_true`:
- What do you think of it ?
- Using directly `y_true` instead of `t_true` is still possible at the moment, should we remove it and only keep support for `t_true` ?
- Do we want to an $\alpha$ factor as in `SPOPlus` ? (currently we can only reproduce `SPOPlus` with $\alpha = 1$)

Reply:
- I don' t like using dispatch for this, it can lead to confusion, for instance if the user is tempted to only provide the real theta. Better to always request a named tuple or do this with optional (keyword?) arguments.
- Adding support for $\alpha$ with a default value of 1 should not be too hard, and it enables more experiments, so I'm in favor

Reply:
- I agree, now in the latest version I pushed, any form of `t_true` is supported, it only needs to have the method `getytrue` implemented
- Ok, support for $\alpha$ should be available now!